frontend/src/composables/useCredentialContext.js

```diff
@@ -0,0 +1,254 @@
+//
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import {
+  ref,
+  shallowRef,
+  computed,
+  inject,
+  provide,
+  watch,
+  watchEffect,
+} from 'vue'
+
+import { useAuthzStore } from '@/store/authz'
+
+import { cleanup } from '@/composables/helper'
+
+import {
+  decodeBase64,
+  encodeBase64,
+} from '@/utils'
+
+import { useObjectMetadata } from './useObjectMetadata'
+
+import cloneDeep from 'lodash/cloneDeep'
+import get from 'lodash/get'
+import isEqual from 'lodash/isEqual'
+import set from 'lodash/set'
+import mapValues from 'lodash/mapValues'
+
+export function createCredentialContextComposable (options = {}) {
```

Comment:
keep useCredentialContext but extract the secretBinding and secret related code into own composables, like `useSecretBindingContext` and `useSecretContext`.